### PR TITLE
8293592: Remove JVM_StopThread, stillborn, and related cleanup

### PR DESCRIPTION
**Pull request by dholmes-ora**
Now that Thread.stop has been degraded to throw `UnsupportedOperationException` (JDK-8299610) the only direct source of async exceptions is from JVMTI `StopThread`. We can remove the `JVM_StopThread` code, remove the `stillborn` field from `java.lang.Thread` and its associated accesses from the VM, and we can stop special-casing `ThreadDeath` handling (as was done for the JDK code as part of JDK-8299610).

Note that JVMTI `StopThread` can only act on a thread that is alive, so it is no longer possible to stop a thread before it has been started.

Also note that there is a change in behaviour for JNI `ExceptionDescribe` as it no longer ignores `ThreadDeath` exceptions (not that it was ever specified to ignore them, it simply mirrored the behaviour of the default `UncaughtExceptionHandler` in `java.lang.ThreadGroup` - which also no longer ignores them so the mirroring behaviour remains the same).

Testing: tiers 1-3

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293592](https://bugs.openjdk.org/browse/JDK-8293592): Remove JVM_StopThread, stillborn, and related cleanup

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)
 * [Robbin Ehn](https://openjdk.org/census#rehn) (@robehn - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10400/head:pull/10400` \
`$ git checkout pull/10400`

Update a local copy of the PR: \
`$ git checkout pull/10400` \
`$ git pull https://git.openjdk.org/jdk pull/10400/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10400`

View PR using the GUI difftool: \
`$ git pr show -t 10400`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10400.diff">https://git.openjdk.org/jdk/pull/10400.diff</a>

</details>

### CODE REVIEW

**Comment by AlanBateman** [comment]
(JDK-8289610 is integrated now so once you sync up it should remove the changes from the dependent PR once the list of changed files. You should be able to trim down the labels too.)

I did a pass over the src/hotspot change and didn't spot any issues. It's good that stillborn can go away. The removal of the special-casing of ThreadDeath is consistent with the libs side of the change, including the change to ExceptionDescribe.

File: src/jdk.jdi/share/classes/com/sun/tools/jdi/VirtualMachineManagerImpl.java [hunk 1]
```diff
@@ -99,12 +99,7 @@ protected VirtualMachineManagerImpl() {
 
             try {
                 connector = connectors.next();
-            } catch (ThreadDeath x) {
-                throw x;
-            } catch (Exception x) {
-                System.err.println(x);
-                continue;
-            } catch (Error x) {
+            } catch (Exception | Error x) {
```
Comment (plummercj) [inline src/jdk.jdi/share/classes/com/sun/tools/jdi/VirtualMachineManagerImpl.java:102]: Maybe this should just catch `Throwable`, although it is unclear to me why we would want to catch any exception here.

File: src/jdk.jdi/share/classes/com/sun/tools/jdi/VirtualMachineManagerImpl.java [hunk 2]
```diff
@@ -128,12 +123,7 @@ protected VirtualMachineManagerImpl() {
 
             try {
                 transportService = transportServices.next();
-            } catch (ThreadDeath x) {
-                throw x;
-            } catch (Exception x) {
-                System.err.println(x);
-                continue;
-            } catch (Error x) {
+            } catch (Exception | Error x) {
```
Comment (plummercj) [inline src/jdk.jdi/share/classes/com/sun/tools/jdi/VirtualMachineManagerImpl.java:126]: Another that could be just catch `Throwable`

File: src/jdk.jdi/share/classes/com/sun/tools/jdi/VirtualMachineManagerImpl.java (hunk 1 above)
Comment (dholmes-ora) [reply src/jdk.jdi/share/classes/com/sun/tools/jdi/VirtualMachineManagerImpl.java]: @plummercj  these changes are not actually part of this PR but came from the upstream PR this was dependent on.

File: src/jdk.jdi/share/classes/com/sun/tools/jdi/VirtualMachineManagerImpl.java (hunk 2 above)
Comment (dholmes-ora) [reply src/jdk.jdi/share/classes/com/sun/tools/jdi/VirtualMachineManagerImpl.java]: @plummercj  these changes are not actually part of this PR but came from the upstream PR this was dependent on.

**Comment by dholmes-ora** [comment]
This PR is now back to its proper form after merging the upstream changes.  I can't say I like the "dependent PR" feature.

File: src/jdk.jdi/share/classes/com/sun/tools/jdi/VirtualMachineManagerImpl.java (hunk 2 above)
Comment (AlanBateman) [reply src/jdk.jdi/share/classes/com/sun/tools/jdi/VirtualMachineManagerImpl.java]: I think this dates from JDK 5 when pluggable JDI connectors and transports were aded. VirtualMachineManager does specify that it tolerates configuration and deployment issues with:

"At start-up time the VirtualMachineManager attempts to load and instantiate (using the no-arg constructor) each class listed in the provider configuration file. Exceptions thrown when loading or creating the Connector are caught and ignored. In other words, the start-up process continues despite of errors."

Looking at it now, I think it should be catching ServiceConfigurationError as that is what the SL's iterator hasNext/next will throw if locating, loading or instantiating a Connector fails. This seems something for an another issue of course, just strange that the Skara bots included it.

**Review by AlanBateman — APPROVED** [review]

**Comment by dholmes-ora** [comment]
Thanks @AlanBateman !

**Review by sspitsyn — APPROVED** [review]
This looks good to me.
I do not see any issues.
Thanks,
Serguei

**Comment by dholmes-ora** [comment]
Thanks @sspitsyn !

**Review by robehn — APPROVED** [review]
Looks good, thank for fixing.

**Review by coleenp — APPROVED** [review]

**Comment by dholmes-ora** [comment]
Thanks @robehn and @coleenp !

/integrate

**Pull request closed without merging**